src/nft/pages/asset/Asset.tsx

```diff
@@ -1,37 +1,289 @@
+  dominantColor: [number, number, number]
+}) => {
+  switch (mediaType) {
+    case 'video':
+      return (
+        <video
+          src={asset.animationUrl}
+          className={styles.image}
+          autoPlay
+          controls
+          muted
+          loop
+          style={{ ['--shadow' as string]: `rgba(${dominantColor.join(', ')}, 0.5)` }}
+        />
+      )
+    case 'image':
+      return (
+        <img
+          className={styles.image}
+          src={asset.imageUrl}
+          alt={asset.name || asset.collectionName}
+          style={{ ['--shadow' as string]: `rgba(${dominantColor.join(', ')}, 0.5)` }}
+        />
+      )
+    case 'audio':
+      return <AudioPlayer {...asset} dominantColor={dominantColor} />
+  }
+}
 
 const Asset = () => {
   const { tokenId = '', contractAddress = '' } = useParams()
-
   const { data } = useQuery(['assetDetail', contractAddress, tokenId], () =>
     fetchSingleAsset({ contractAddress, tokenId })
   )
+  const { pathname, search } = useLocation()
+  const navigate = useNavigate()
+  const bagExpanded = useBag((state) => state.bagExpanded)
+  const [creatorAddress, setCreatorAddress] = useState('')
+  const [ownerAddress, setOwnerAddress] = useState('')
+  const [dominantColor] = useState<[number, number, number]>([0, 0, 0])
+  const creatorEnsName = useENSName(creatorAddress)
+  const ownerEnsName = useENSName(ownerAddress)
+  const parsed = qs.parse(search)
+  const asset = useMemo(() => (data ? data[0] : ({} as GenieAsset)), [data])
+  const collection = useMemo(() => (data ? data[1] : ({} as CollectionInfoForAsset)), [data])
+  const { gridWidthOffset } = useSpring({
+    gridWidthOffset: bagExpanded ? 324 : 0,
+  })
 
-  let asset = {} as GenieAsset
-  let collection = {} as CollectionInfoForAsset
+  useEffect(() => {
+    if (asset.creator) setCreatorAddress(asset.creator.address)
+    if (asset.owner) setOwnerAddress(asset.owner)
+  }, [asset])
 
-  if (data) {
-    asset = data[0] || {}
-    collection = data[1] || {}
-  }
+  const { rarityProvider } = useMemo(
+    () =>
+      asset.rarity
+        ? {
+            rarityProvider: asset.rarity.providers.find(
+              ({ provider: _provider }) => _provider === asset.rarity?.primaryProvider
+            ),
+            rarityLogo: rarityProviderLogo[asset.rarity.primaryProvider] || '',
+          }
+        : {},
+    [asset.rarity]
+  )
+
+  const assetMediaType = useMemo(() => {
+    if (isAudio(asset.animationUrl)) {
+      return 'audio'
+    } else if (isVideo(asset.animationUrl)) {
+      return 'video'
+    }
+    return 'image'
+  }, [asset])
 
   return (
-    <div>
-      {' '}
-      <Details
-        contractAddress={contractAddress}
-        tokenId={tokenId}
-        tokenType={asset.tokenType}
-        blockchain="Ethereum"
-        metadataUrl={asset.externalLink}
-        totalSupply={collection.totalSupply}
-      />
-    </div>
+    <AnimatedBox
+      style={{
+        // @ts-ignore
```

Comment:
what is the ignore for?

Reply:
Animated box is ```animated(Box)```, it doesn't return the correct types of some react-spring type.  I tried fixing the types, but it broke stuff elsewhere.  So, the easiest thing was to simply ignore it for now.